broken parmetis link fix

lepotatoguy opened the pull request:
Previous link of Parmetis 4.0.2 is invalid/broken. I have updated it with the latest working link of Parmetis v4.0.2.

showrav-ansary commented:
+1

rupertnash commented on geometry-tool/setup.py:
```diff
@@ -45,11 +45,10 @@
     python_requires=">=3.6",
     install_requires=[
         "pyyaml",
-        # Numpy >= 1.20 requires python 3.7; VMTK conda binaries are 3.6 only
```
Leave comment

rupertnash commented on README.md:
```diff
@@ -29,4 +29,4 @@ Key publications:
   https://doi.org/10.1103/PhysRevE.89.023303
 
-Please see the [doc] folder for more details.
+Documentation available at the [doc](/doc/) folder and [here](https://docs.google.com/document/d/1_3WR3MR7mFyE9LxzcSeXy--G3qgnA3TBxO8aDqUDW2Q/edit?usp=sharing).
```
Remove link to google doc

rupertnash commented on doc/user/python-tools.md:
```diff
@@ -32,7 +32,7 @@ Available command line tools:
 - `hlb-gmy-compress`: compress an uncompressed geometry file.
 
-- `hlb-dump-extracted-properties`: convert an extraction file to CSV.
+- `hlb-dump-extracted-properties`: convert an extraction file (.xtr) to CSV. (Instead of HemeXtract, HemeXtract is obsolete for this version of HemeLB.)
```
Rephrase: Obsolete implies HemeXtract was once current for this repo. It was developed for one of the forks.

rupertnash commented on geometry-tool/setup.py:
```diff
@@ -45,11 +45,10 @@
         "numpy; python_version >= '3.7'",
         "vtk ~= 9.0",
-        "vmtk ~= 1.5",
+        #"vmtk ~= 1.5",
```
Leave VMTK req

lepotatoguy replied:
If I do this, it shows an error when installing HemeLB even though vmtk was installed as a conda package. Traceback: 

Requirement already satisfied: vtk~=9.0 in /home/joyanta-mondal/anaconda3/envs/gmy-tool-fresh/lib/python3.8/site-packages (from HlbGmyTool==1.2) (9.1.0)
INFO: pip is looking at multiple versions of hlbgmytool to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement vmtk~=1.5 (from hlbgmytool) (from versions: none)
ERROR: No matching distribution found for vmtk~=1.5

rupertnash replied:
Yes, because it's a requirement.

Some conda installs of packages including VMTK are not visible to setuptools because of *reasons*.

We used to have a script to bodge this (https://github.com/hemelb-codes/hemelb/blob/e29ff7aafeffe120d84413961dd6c752f7e10417/geometry-tool/bodge-packages-for-setuptools.sh) but I had thought newer versions had resolved this

lepotatoguy replied:
Okay. Thank you for the clarification.

rupertnash commented on hemelb-spec-2024-12-06.txt:
Remove this

rupertnash reviewed (changes requested):
Thanks for the help.

rupertnash commented:
I see that this is a known problem due to the Karypis group moving their website to Github 
https://github.com/KarypisLab/ParMETIS/issues/27

The replacement URL should be https://karypis.github.io/glaros/files/sw/parmetis/parmetis-4.0.2.tar.gz rather than the CP2K website

Thanks for this.

lepotatoguy committed:
fix broken parmetis link and update python dependencies

add hemelb python dependencies

remove vmtk requirement, because this can be covered through hemelb-spec-2024-12-06.txt file

update vagrantfile

update gitignore

update vmtk

update gitignore

lepotatoguy commented:
> I see that this is a known problem due to the Karypis group moving their website to Github [KarypisLab/ParMETIS#27](https://github.com/KarypisLab/ParMETIS/issues/27)
> 
> The replacement URL should be https://karypis.github.io/glaros/files/sw/parmetis/parmetis-4.0.2.tar.gz rather than the CP2K website
> 
> Thanks for this.

Updated the link. Thank you.

lepotatoguy commented:
> Can you please rebase and squash the commits into 2 or 3? Thanks!

Yes. Done from my side. Thank you.

lepotatoguy commented:
I have pushed the commit [ff6f3ae](https://github.com/hemelb-codes/hemelb/commit/ff6f3aef078a6a325995dd09c5247283b054cdc3), where you updated GH action version, in my fork.

lepotatoguy committed:
update documentation

lepotatoguy committed:
update spelling mistake

rupertnash commented:
Hi - thanks for spotting this and making the change. There are a whole load of minor edits to things that I can't accept (e.g the reflowing of large bits of the docs). I have cherry picked to main the change at the crux of this PR. If you wish to do a documentation tidy that would be gratefully accepted but needs to not make large amounts on unneeded and undesired changes. Thanks

lepotatoguy commented:
> Hi - thanks for spotting this and making the change. There are a whole load of minor edits to things that I can't accept (e.g the reflowing of large bits of the docs). I have cherry picked to main the change at the crux of this PR. If you wish to do a documentation tidy that would be gratefully accepted but needs to not make large amounts on unneeded and undesired changes. Thanks

@rupertnash Thank you for your earlier feedback.

I’ve submitted a new PR with only minimal and relevant documentation updates, available here: https://github.com/hemelb-codes/hemelb/pull/824. It only changes to essential clarifications, structural fixes, and new machine-specific content.

Please let me know if any revisions are needed. 

Thanks!
